civis/ml/_model.py

```diff
@@ -801,16 +885,14 @@ def __getstate__(self):
     def __setstate__(self, state):
         self.__dict__ = state
         self._client = APIClient()
-        template_ids = self._get_template_ids(self._client)
-        self.train_template_id, self.predict_template_id = template_ids
```

Comment:
Before this PR: The behavior was that the code always tried to get the latest template IDs, which might be different from before the pickled `ModelPipeline` was created (due to a civis-python/CivisML new version).

After this PR: When a `ModelPipeline` instance is created, the training template ID is already set, possibly by the user-provided `civisml_version` parameter. This means the unpickling a `ModelPipeline` object shouldn't attempt to update the training and prediction template IDs anymore. Hence deleting these two lines. I've added a test to make sure that template IDs are preserved before vs after pickling.